Framework/DataObjects/inc/MantidDataObjects/TableColumn.h

```diff
@@ -162,7 +162,17 @@ template <class Type> class TableColumn : public API::Column {
         typename boost::mpl::if_c<boost::is_convertible<double, Type>::value,
                                   Type, InconvertibleToDoubleType>::type
             DoubleType;
-    return boost::numeric_cast<double, DoubleType>(m_data[i]);
+    if (typeid(m_data[i]) == typeid(std::string)) {
+      std::string trimmedString = boost::lexical_cast<std::string>(
+          m_data[i]); // so not to alter the original data
+
+      if (trimmedString.back() == ':') {
```

Comment:
1. I agree there is no need for the lexical cast here. The element is already a string. 
2. As I understand it the fix addresses also the issue where the string cannot be converted because of a trailing colon. This colon is removed here and the string is further processed.